### ISSUE TITLE
TLS record HMAC 3: remove non-PSA keys

### ISSUE THREAD

**Issue by mpg** [ARMmbed/mbedtls#5179]
This is the conclusion of a series starting with #5176 and continuing with #5177 and #5178. Now that all computations use PSA, we can remove the legacy contexts from the transform when `MBEDTLS_USE_PSA_CRYPTO` is defined.

- [x] In `struct mbedtls_ssl_transform`, remove `md_ctx_enc`/`md_ctx_dec`  when `MBEDTLS_USE_PSA_CRYPTO` is defined.
- [x] In `mbedtls_ssl_transform_init()` and `mbedtls_ssl_transform_free()`, remove related code  when `MBEDTLS_USE_PSA_CRYPTO` is defined.
- [x] In `ssl_tls12_populate_transform()`, code using those fields  when `MBEDTLS_USE_PSA_CRYPTO` is defined.

Depends on: #5177 and #5178 - for the legacy context to be no longer needed.

**Comment by superna9999** [comment]
@mpg While implementing this, I tried to drop `md_info` struct & MBedTLS api call, but I was blocked by the removal of:
```c
mac_key_len = mbedtls_md_get_size( md_info );
```
I couln't find any way to determine the MAC key length using PSA api, is this planned ?

**Comment by superna9999** [comment]
While #5573 implements this, I think it could go further to a complete removal of `md_info` in `ssl_tls12_populate_transform()`, should I create a new issue for that or put a new PR when #5573 is merged ?

**Comment by mpg** [comment]
Hmm, good question. Strictly speaking, I think that step further would be part of #5210 - remove the remaining dependencies on MD - which we weren't planning on working on right now. OTOH, it's always better to strike the iron while it's hot :) How much work do you think it would be? If it's just one or two smallish commits, perhaps just do it as part of 5573? Otherwise, I'd say open a PR after 5573 is merged. Or you could also just leave a note in 5210 with your thoughts. Up to you!

**Comment by superna9999** [comment]
It's a small commit, I'll open a PR after #5573 is merged, not need to add more complexity to it.
